Refactor: Extract utility functions into separate modules

SameDesu123 opened the pull request:
Extracted various utility functions and modal management logic from src/index.js into new, dedicated modules: src/imageUtils.js, src/modalManager.js, and src/utils.js. This improves code organization, modularity, and maintainability by separating concerns. Updated all references and imports in src/index.js accordingly.

SameDesu123 committed:
Refactor: Extract utility functions into separate modules

Extracted various utility functions and modal management logic from src/index.js into new, dedicated modules: src/imageUtils.js, src/modalManager.js, and src/utils.js. This improves code organization, modularity, and maintainability by separating concerns. Updated all references and imports in src/index.js accordingly.

SameDesu123 committed:
Fix: Address PR feedback on utility functions and imports

Addressed feedback from PR #5:
- Imported showInfoModal in src/imageUtils.js to resolve runtime errors.
- Refactored handleSaveCharacterToImage in src/imageUtils.js to receive characterData as an argument, decoupling it from direct DOM access. Updated its call in src/index.js accordingly.
- Removed unused 'language' import from src/modalManager.js.

concertypin committed:
Change arrow functions to named functions

concertypin committed:
Document with JSDoc

concertypin commented on src/imageUtils.js:
```diff
@@ -0,0 +1,219 @@
+            img.src = event.target.result;
+            img.onload = () => {
+                const canvas = document.createElement('canvas');
+                let width = img.width;
+                let height = img.height;
+
+                if (width > height) {
+                    if (width > maxWidth) {
+                        height *= maxWidth / width;
+                        width = maxWidth;
+                    }
+                } else {
+                    if (height > maxHeight) {
+                        width *= maxHeight / height;
+                        height = maxHeight;
+                    }
+                }
```
`async` keyword indicates this is async function explicitly. I think this should be remained.

concertypin commented:
Good commit! Please inform me whether you will apply these suggestions.

SameDesu123 commented on src/imageUtils.js:
```diff
@@ -0,0 +1,219 @@
+                const jsonString = decodeTextFromImage(imageData);
+                if (jsonString) {
+                    const data = JSON.parse(jsonString);
+                    if (data.source === 'PersonaChatAppCharacterCard') {
```
Good call. Using enums for the source will make it much easier to read. I'm for it.

SameDesu123 committed:
Refactor: Use constant for character card source ID

Replaced the hardcoded string PersonaChatAppCharacterCard with a constant from an enum-like object (SourceTypes.CHARACTER_CARD).

This improves readability and maintainability by centralizing the value, making future updates easier and preventing potential typos.

SameDesu123 commented:
I think we should rename the src/defauts.js file later. It looks like there's a typo and it should be defaults.js.

SameDesu123 commented:
I'll repost the PR again after checking the new commit.
